src/animation.js

```diff
@@ -130,13 +132,39 @@
         return 'pending';
       if (this.paused)
         return 'paused';
-      if (this.finished)
+      if (this._isFinished)
         return 'finished';
       return 'running';
     },
+    get finished() {
+      if (!this._finishedPromise) {
```

Comment:
I made them lazy because the Blink impl does (as far as I can tell). We only need to check for promise changes at state transitions if the promise exists.